viadot/sources/sharepoint.py

```diff
@@ -52,7 +52,8 @@ def download_file(
         download_to_path: str = "Sharepoint_file.xlsm",
     ) -> None:
 
-        if not download_from_path or self.download_from_path:
+        download_from_path = download_from_path or self.download_from_path
```

Comment:
If you want to keep the path as an attribute (so it can be used in eg. self.to_df()), I would call it just `path` instead (as it can be used for other things than downloading). So, in `download_file()`, you would have: `download_from_path = download_from_path or self.path`. Otherwise, deleting this attribute completely would also be fine since it's only used in `download_file()`.